### PR TITLE
Parsing team

### PR DESCRIPTION
**Pull request by apapuc97**
Added entities, repos and services

Also added sql scripts for creating the data base schema and the user for accessing it.

### CODE REVIEW

File: src/main/java/com/ness/breakdownthecode/service/EntityService.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.ness.breakdownthecode.service;
+
+import com.ness.breakdownthecode.entity.GitEntity;
+
+import java.util.List;
+
+public interface EntityService {
```
Comment (ncsereoka) [inline src/main/java/com/ness/breakdownthecode/service/EntityService.java:7]: I think we should rename the 'Entity' table/class to something else.
We cause unnecessary confusion by having a GitEntity class along with an EntityService and an Entity table.
File would be a more pertinent name, but that's a reserved class from JDK.
I suggest Record or Source (or Document, though not as good as the first two). Other opinions?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1 +1,3 @@
-
+spring.datasource.url=jdbc:mysql://localhost:3306/break_down_the_code?useSSL=false&serverTimezone=UTC
+spring.datasource.username=ness
+spring.datasource.password=ness
```
Comment (ncsereoka) [inline src/main/resources/application.properties:3]: Is this alright for now? Or should we remove passwords and such from source control?

File: src/main/java/com/ness/breakdownthecode/entity/Change.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import lombok.Setter;
+
+import javax.persistence.*;
+
+/**
+ * The java entity class for 'change' table
+ */
+@Entity
+@Table(name = "change")
+public class Change
```
Comment (nfazekas) [inline src/main/java/com/ness/breakdownthecode/entity/Change.java:14]: In loc de @Getter @Setter pe fiecare variabile, poti sa pui @Data pe clasa

File: src/main/java/com/ness/breakdownthecode/entity/Contribuitor.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.persistence.*;
+
+/**
+ * The java entity class for 'contribuitor' table
+ */
+@Entity
```
Comment (nfazekas) [inline src/main/java/com/ness/breakdownthecode/entity/Contribuitor.java:11]: In loc de @Getter @Setter pe fiecare variabile, poti sa pui @Data pe clasa

File: src/main/java/com/ness/breakdownthecode/entity/GitEntity.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import lombok.Setter;
+
+import javax.persistence.*;
+
+/**
+ * The java entity class for 'entity' table
+ */
+@Entity
+@Table(name = "entity")
+public class GitEntity {
```
Comment (nfazekas) [inline src/main/java/com/ness/breakdownthecode/entity/GitEntity.java:13]: In loc de @Getter @Setter pe fiecare variabile, poti sa pui @Data pe clasa

File: src/main/java/com/ness/breakdownthecode/service/ChangeServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    @Autowired
+    private ChangeRepository changeRepository;
+
+    @Override
+    public List<Change> findAll() {
+        return changeRepository.findAll();
+    }
+
+    @Override
+    public Change findById(int id) {
```
Comment (nfazekas) [inline src/main/java/com/ness/breakdownthecode/service/ChangeServiceImpl.java:23]: Mai bine returnzezi Optional decat null

File: src/main/java/com/ness/breakdownthecode/service/ContribuitorServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    @Autowired
+    private ContribuitorRepository contribuitorRepository;
+
+    @Override
+    public List<Contribuitor> findAll() {
+        return contribuitorRepository.findAll();
+    }
+
+    @Override
+    public Contribuitor findById(int id) {
```
Comment (nfazekas) [inline src/main/java/com/ness/breakdownthecode/service/ContribuitorServiceImpl.java:25]: La fel, prefer Optional decat null.

File: src/main/resources/application.properties (hunk 2 above)
Comment (nfazekas) [inline src/main/resources/application.properties:3]: Puteti configura ca Spring sa va creeze automat tabelele. Voi ar trebui sa creati numai schema, tabelele sa fie create de Spring.

File: src/main/java/com/ness/breakdownthecode/service/EntityService.java (hunk 1 above)
Comment (lalasmurf) [reply src/main/java/com/ness/breakdownthecode/service/EntityService.java:7]: Ok. As long as you are consistent with the naming. Pick one of them Record/Document/GitRecord/anything suggestive.

**Pull request merged**